src/UXClient/Components/SingleDateTimePicker/SingleDateTimePicker.ts

```diff
@@ -202,10 +202,22 @@ class SingleDateTimePicker extends ChartComponent{
         return Utils.parseUserInputDateTime(this.timeInput.node().value, this.chartOptions.offset);
     }
 
-    private updateDisplayedDateTime () {
-        var selectedDate = new Date(this.millis);
-        this.calendarPicker.setDate(this.roundDay(Utils.offsetFromUTC(selectedDate)));
-        this.timeInput.node().value = this.createTimeString(Utils.offsetFromUTC(selectedDate));
+    private offsetFromUTC (date: Date) {
+        let dateCopy = new Date(date.valueOf())
+        dateCopy.setTime(dateCopy.getTime() + dateCopy.getTimezoneOffset()*60*1000 );
+        return dateCopy;    
+    }
+
+    private convertToCalendarDate (millis: number) {
+        return this.roundDay(this.offsetFromUTC(Utils.offsetFromUTC(new Date(millis), this.chartOptions.offset)))
```

Comment:
nit: it's a little confusing how these two methods are both called offsetFromUTC. Like if you call the version from Utils it doesn't do what you need so you need to call another one. Maybe you could rename the one here to reflect why it needs to modify the result from Utils.offsetFromUTC? Up to you.

Reply:
You're right - the two methods actually have different behavior so I should change the name of one of them